bundles/org.openhab.binding.pulseaudio/src/main/java/org/openhab/binding/pulseaudio/internal/PulseaudioClient.java

```diff
@@ -237,114 +242,59 @@ public void sendCommand(String command) {
-     *
-     * @return the corresponding {@link SinkInput} to the given <code>id</code>
-     */
-    public @Nullable SinkInput getSinkInput(int id) {
-        for (AbstractAudioDeviceConfig item : items) {
-            if (item.getId() == id && item instanceof SinkInput) {
-                return (SinkInput) item;
-            }
-        }
-        return null;
-    }
-
-    /**
-     * retrieves a {@link Source} by its name
-     *
-     * @return the corresponding {@link Source} to the given <code>name</code>
-     */
-    public @Nullable Source getSource(String name) {
-        for (AbstractAudioDeviceConfig item : items) {
-            if (item.getPaName().equalsIgnoreCase(name) && item instanceof Source) {
-                return (Source) item;
-            }
-        }
-        return null;
-    }
-
     /**
      * retrieves a {@link Source} by its id
      *
      * @return the corresponding {@link Source} to the given <code>id</code>
      */
     public @Nullable Source getSource(int id) {
-        for (AbstractAudioDeviceConfig item : items) {
+        for (AbstractAudioDeviceConfig item : getItems()) {
             if (item.getId() == id && item instanceof Source) {
                 return (Source) item;
             }
         }
         return null;
     }
 
-    /**
-     * retrieves a {@link SourceOutput} by its name
-     *
-     * @return the corresponding {@link SourceOutput} to the given <code>name</code>
-     */
-    public @Nullable SourceOutput getSourceOutput(String name) {
-        for (AbstractAudioDeviceConfig item : items) {
-            if (item.getPaName().equalsIgnoreCase(name) && item instanceof SourceOutput) {
-                return (SourceOutput) item;
-            }
-        }
-        return null;
-    }
-
-    /**
-     * retrieves a {@link SourceOutput} by its id
-     *
-     * @return the corresponding {@link SourceOutput} to the given <code>id</code>
-     */
-    public @Nullable SourceOutput getSourceOutput(int id) {
-        for (AbstractAudioDeviceConfig item : items) {
-            if (item.getId() == id && item instanceof SourceOutput) {
-                return (SourceOutput) item;
-            }
-        }
-        return null;
-    }
-
     /**
      * retrieves a {@link AbstractAudioDeviceConfig} by its name
      *
      * @return the corresponding {@link AbstractAudioDeviceConfig} to the given <code>name</code>
      */
     public @Nullable AbstractAudioDeviceConfig getGenericAudioItem(String name) {
-        for (AbstractAudioDeviceConfig item : items) {
+        for (AbstractAudioDeviceConfig item : getItems()) {
             if (item.getPaName().equalsIgnoreCase(name)) {
                 return item;
             }
         }
         return null;
     }
 
+    /**
+     * Get all items previously parsed from the pulseaudio server.
+     * If no scan has occured, then wait an extra 2 seconds before retrying.
+     * (the wait can prevent error on startup)
```

Comment:
We have no guarantee that the update method is already called (it could be already called, but maybe it _will_ be called in  ten millisecond), so a synchronized alone would not be sufficient.
We could do a synchronize, _and_ call the update method from inside getItems when items are null, but it will, in a case with several pulseaudiohandler under the same bridge, make (at start) numerous unnecessary and subsequent call to getItems() and update() (one per pulseaudiohandler initialize) when the lock will be freed and attributed to each device sequentially.
I'm not pleased with the wait, but I thought it was the workaround with the less code complexity and the more readability.
But as it's clearly not elegant, I can do with a lock and call, if you would like ?
(Or if you have another idea ?)

Reply:
I upgrade myself on the subject of synchronization/wait, and I think I managed to do what you want in my last commit.
The child handlers now wait for the bridge handler completing its initialization.
They stop and are notified when so.

Reply:
The method description has then to be updated

Reply:
Indeed, sorry !